### PR TITLE
Update Microsoft.Extensions packages

### PR DESCRIPTION
**Pull request by ryuyu**
Update references to Microsoft.Extensions packages from 1.x -> 2.x
Also retargets net472

Depends on https://github.com/NuGet/ServerCommon/pull/318

### CODE REVIEW

File: src/Catalog/NuGet.Services.Metadata.Catalog.csproj [hunk 1]
```diff
@@ -291,7 +291,7 @@
       <Version>4.4.5-dev-2939401</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Sql">
-      <Version>2.57.0</Version>
+      <Version>2.58.0-ryuyu-update-common-3142384</Version>
```
Comment (agr) [inline src/Catalog/NuGet.Services.Metadata.Catalog.csproj]: Please, use a proper release.

File: src/Ng/Telemetry/JobNameTelemetryInitializer.cs [hunk 2]
```diff
@@ -23,8 +23,8 @@ public JobNameTelemetryInitializer(string jobName, string instanceName)
 
         public void Initialize(ITelemetry telemetry)
         {
-            telemetry.Context.Properties[JobNameKey] = _jobName;
-            telemetry.Context.Properties[InstanceNameKey] = _instanceName;
+            telemetry.Context.GlobalProperties[JobNameKey] = _jobName;
```
Comment (joelverhagen) [inline src/Ng/Telemetry/JobNameTelemetryInitializer.cs]: I think the proper way is like this as @xavierdecoster found:
https://github.com/NuGet/NuGetGallery/blob/665c759cd739252134c654e920c1ac6a5aecff72/src/NuGetGallery/Telemetry/ClientInformationTelemetryEnricher.cs#L29

File: src/NuGet.Services.BasicSearch/DeploymentIdContextInitializer.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ public void Initialize(ITelemetry telemetry)
         {
             if (SafeRoleEnvironment.TryGetDeploymentId(out var id))
             {
-                telemetry.Context.Properties[DeploymentId] = id;
+                telemetry.Context.GlobalProperties[DeploymentId] = id;
```
Comment (joelverhagen) [inline src/NuGet.Services.BasicSearch/DeploymentIdContextInitializer.cs]: Same as above:
https://github.com/NuGet/NuGetGallery/blob/665c759cd739252134c654e920c1ac6a5aecff72/src/NuGetGallery/Telemetry/ClientInformationTelemetryEnricher.cs#L29

File: src/NuGet.Services.BasicSearch/Web.config [hunk 4]
```diff
@@ -66,6 +66,10 @@
   </system.webServer>
   <runtime>
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+			<dependentAssembly>
```
Comment (joelverhagen) [inline src/NuGet.Services.BasicSearch/Web.config]: nit: consistent indentation.

might be worth deleting this whole `<runtime>` section and regenerating it from scratch using that build warning in VS.

File: src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs [hunk 5]
```diff
@@ -33,7 +35,31 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
...
         {
-            serializer.Serialize(writer, value);
+            var ex = value as Exception;
+            var serializableEx = new Wrapper(ex);
+            serializer.Serialize(writer, serializableEx);
+        }
+
+        ///<remarks>
+        /// This class needs to exist because System.Exception is not marked with as Serializable attribute
+        ///  and old JSON.NET behaviour was incorrectly treating all ISerializable as Serializable
```
Comment (joelverhagen) [inline src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs]: nit: there is an extra space before `and old`

File: src/NuGet.Services.SearchService/Web.config [hunk 6]
```diff
@@ -49,23 +49,23 @@
       </dependentAssembly>
       <dependentAssembly>
         <assemblyIdentity name="System.Runtime" publicKeyToken="B03F5F7F11D50A3A" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-4.1.1.0" newVersion="4.1.1.0"/>
+        <bindingRedirect oldVersion="0.0.0.0-4.1.2.0" newVersion="4.1.2.0"/>
```
Comment (joelverhagen) [inline src/NuGet.Services.SearchService/Web.config]: Consider regenerating from scratch to ensure it is minimal, as mentioned above with BasicSearch.

File: src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs (hunk 5 above)
Comment (ryuyu) [reply src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs]: done

File: src/NuGet.Services.SearchService/Web.config (hunk 6 above)
Comment (ryuyu) [reply src/NuGet.Services.SearchService/Web.config]: regenerated

File: src/NuGet.Services.BasicSearch/Web.config (hunk 4 above)
Comment (ryuyu) [reply src/NuGet.Services.BasicSearch/Web.config]: Fixed

File: src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs [hunk 7]
```diff
@@ -33,7 +35,31 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
...
+            var ex = value as Exception;
+            var serializableEx = new Wrapper(ex);
+            serializer.Serialize(writer, serializableEx);
+        }
+
+        ///<remarks>
+        /// This class needs to exist because System.Exception is not marked with as Serializable attribute
+        /// and old JSON.NET behaviour was incorrectly treating all ISerializable as Serializable
+        /// This was changed between 10.x and 11.x. See https://github.com/JamesNK/Newtonsoft.Json/issues/1622 for more details
+        /// For our purposes, passing through the GetObjectData call from a custom class marked as serialable is sufficient
```
Comment (loic-sharma) [inline src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs]: >serialable [](start = 99, length = 10)

serializable

**Pull request merged**